app/plugin_api.go

```diff
@@ -109,7 +109,7 @@ func (api *PluginAPI) GetChannel(channelId string) (*model.Channel, *model.AppEr
 }
 
 func (api *PluginAPI) GetChannelByName(name, teamId string) (*model.Channel, *model.AppError) {
-	return api.app.GetChannelByName(name, teamId)
+	return api.app.GetChannelByName(name, teamId, false)
```

Comment:
Makes sense to include here the IncludeDeleted parameter or a new GetChannelByNameIncludeDeleted function? Plugins can access to the Archive Channels too?
//cc @jwilander

Reply:
The consistency argument would say to keep it the same as the other API.

Reply:
Sorry, missed the mention. Chatting directly with @jespino

Reply:
I asked @jwilander and the proposal is to use Option pattern, and in a new PR to review the result.

I going to create a new PR for that.